src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/Versioning/ControlPlaneCollectionUpgradeManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Documents;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.CosmosDb.Configs;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Microsoft.Health.CosmosDb.Features.Storage.Versioning;
+
+namespace Microsoft.Health.ControlPlane.CosmosDb.Features.Storage.Versioning
+{
+    public class ControlPlaneCollectionUpgradeManager : IUpgradeManager
+    {
+        private readonly IEnumerable<ICollectionUpdater> _collectionUpdater;
+        private readonly CosmosDataStoreConfiguration _configuration;
+
+        /// <summary>
+        /// This integer should be incremented when changing any value in the
+        /// UpdateIndexAsync function
+        /// </summary>
+        internal const int CollectionSettingsVersion = 1;
+
```

Comment:
nit: i would move these up with the two entries above.

Reply:
Moved.